cdk/stack.py

```diff
@@ -240,20 +241,24 @@ def build_api_lambda(
                 )
             ],
         )
-        handler = aws_lambda_python_alpha.PythonFunction(
+        handler = aws_lambda.Function(
             self,
             "api-handler",
-            entry="api",
-            index="src/handler.py",
+            code=aws_lambda.Code.from_docker_build(
+                path=os.path.abspath(code_dir),
+                file="api/Dockerfile",
+                platform="linux/amd64",
+            ),
             runtime=aws_lambda.Runtime.PYTHON_3_9,
             timeout=Duration.seconds(30),
+            handler="handler.handler",
             role=handler_role,
             environment={"DB_SECRET_ARN": db_secret.secret_arn, **env},
             vpc=db_vpc,
             vpc_subnets=ec2.SubnetSelection(
                 subnet_type=ec2.SubnetType.PUBLIC
                 if db_subnet_public
-                else ec2.SubnetType.PRIVATE_WITH_NAT
+                else ec2.SubnetType.PRIVATE_ISOLATED
```

Comment:
This addresses a different deployment issue - `PRIVATE_ISOLATED` is the subnet type we currently use in the `veda-backend` VPC.